### PR TITLE
Ensure tokens revoked from Cognito aren't used to interact with Fractl

### PR DESCRIPTION
**Pull request by pratik-fractl**
Keep track of UserSessions and safe guard against Users who have logged out from Design Studio yet, use the JWT token via HTTP requests.

### CODE REVIEW

File: src/fractl/http.clj [hunk 1]
```diff
@@ -947,34 +956,42 @@
...
+  (let [user (get-in request [:identity :sub])
+        logged-in-status (get
+                           (first
+                             (get
+                               (first
+                                 (evaluator
+                                   {:Fractl.Kernel.Identity/FindUserSessionInfo
+                                    {:User user}})) :result)) :LoggedIn)]
+    (try
+      (when-not (and (buddy/authenticated? request) (= logged-in-status true))
```
[the remark this answers is not in the document]
Comment (pratik-fractl) [reply src/fractl/http.clj:969]: Acutally, `buddy/authenticated?` doesn't decode the jwt. It just checks for `:identity` key present in the request.

File: src/fractl/http.clj [hunk 2]
```diff
@@ -947,34 +956,42 @@
...
-  (try
-    (when-not (buddy/authenticated? request)
-      (log/info (str "unauthorized request - " request))
-      (unauthorized (find-data-format request)))
-    (catch Exception ex
-      (log/warn ex)
-      (bad-request "invalid auth data" (find-data-format request)))))
+(defn- handle-request-auth [request evaluator]
+  (let [user (get-in request [:identity :sub])
+        logged-in-status (get
```
[the remark this answers is not in the document]
Comment (pratik-fractl) [reply src/fractl/http.clj:961]: Will do.

File: src/fractl/http.clj [hunk 3]
```diff
@@ -622,7 +622,23 @@
                           (assoc
                            auth-config
                            :event evobj
-                           :eval evaluator))]
+                           :eval evaluator))
+                  user-id (get (jwt/decode (get-in result [:authentication-result :id-token])) :sub)]
+              (if
```
Comment (vijayfractl) [inline src/fractl/http.clj:627]: Please move the session event evaluations to a separate fn and reuse that here and at line 801.

File: src/fractl/http.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (vijayfractl) [reply src/fractl/http.clj:988]: even then `(= logged-in-status true)` is cheaper.

**Review by vijayfractl — CHANGES_REQUESTED** [review]

**Review by vijayfractl — CHANGES_REQUESTED** [review]

**Review by vijayfractl — APPROVED** [review]

**Pull request merged**